src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/PrivilegesInterceptorImpl.java

```diff
@@ -203,6 +224,7 @@ private void replaceIndex(final ActionRequest request, final String oldIndexName
                 }
 
                 if (ar instanceof IndexRequest) {
+                    createIndexRequest = checkAndCreateRequestIfAbsent(newIndexName);
                     ((IndexRequest) ar).index(newIndexName);
                 }
 
```

Comment:
Was not able to add comment to the below code (see below 3 lines).

Should we create the kibana+tenant index for UpdateRequest as well?

Reply:
My understanding of UpdateRequest is that it tries to update an existing document. It means that both document and index should already exist.